fulcio-spec.md

```diff
@@ -0,0 +1,307 @@
+
+A credential SHOULD be a JSON Web Token (JWT) OpenID Connect ID token that conforms to the [OpenID Connect specification](https://openid.net/specs/openid-connect-core-1_0.html#IDToken). The JWT MUST contain:
+
+   * `aud`, the audience of the token, set to \`sigstore\`
+   * `iss`, the issuer of the token
+   * `exp`, the expiration of the token
+   * `iat`, the time of token issuance
+
+The JWT must also include a claim to represent user identity. This is configurable for each supported provider. Common claims include `sub` and `email_address`.
+
+#### 2.2.2. Machine Identity
+
+Fulcio issues certificates not only for user-based identifiers but also machine identities. These can be specified through [SPIFFE IDs](https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/#spiffe-id), Kubernetes service accounts, or CI workflow identifiers. These identifiers will include information about the environment that the workflow occurred in. This could include the Kubernetes cluster, Cloud project, or source repository.
+
+Using machine identities instead of human identifiers provides privacy benefits, as no user identity is associated with the signing event. It also can provide a stronger link between source and build if a machine identity from a CI workflow signs a build.
+
+See [Spec: Fulcio](https://docs.google.com/document/d/1W5xp3g8_jaqzDQmIvepNYsWb-bQNc0U2ZQgQ700Kjok/edit#heading=h.lrl46qy6806e) for more information about required certificate extensions.
+
+### 2.3. Keys
+
+Note that Fulcio can be used with self-managed keys, as described in a [Sigstore blog post](https://blog.sigstore.dev/adopting-sigstore-incrementally-1b56a69b8c15). Fulcio does not enforce the use of ephemeral keys. The signer may have experience with managing keys and appropriately locking down the signing key, for example through the use of an HSM or ACLs. Fulcio still provides benefits through a tighter integration with Sigstore and simplifying the verification policy.
+
+## 3. Issuance - Life of a Request
+
+The client submits a certificate request to Fulcio. The certificate request MUST contain a certificate request described subsequently and OpenID Connect (OIDC) identity token. This is a signed JWT containing information about the principal (identity of the client), the issuer (who issued the identity token - Google, Microsoft, GitHub, etc.) and additional metadata such as expiration. The principal identity can either be a signing identity in the form of an email or username, or a workload identity. The certificate request MUST contain either:
```

Comment:
Do you mean "identity of the signer" instead of "identity of the client"? 'client' has been consistently used to refer to the signing/verification tool

Reply:
Yes, I like `signer` more, that is more precise. I overload `client` to refer to the signer.